esphome/components/cd74hc4067/cd74hc4067.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include "esphome/core/component.h"
+#include "esphome/core/hal.h"
+#include "esphome/components/sensor/sensor.h"
+#include "esphome/components/voltage_sampler/voltage_sampler.h"
+
+namespace esphome {
+namespace cd74hc4067 {
+
+class CD74HC4067Component : public Component {
+ public:
+  /// Set up the internal sensor array.
+  void setup() override;
+  void dump_config() override;
+  float get_setup_priority() const override;
+
+  /// setting pin active by setting the right combination of the four multiplexer input pins
+  void activate_pin(uint8_t pin);
+
+  /// set the pin connected to multiplexer control pin 0
+  void set_pin_s0(InternalGPIOPin *pin) { this->pin_s0_ = pin; }
+  /// set the pin connected to multiplexer control pin 1
+  void set_pin_s1(InternalGPIOPin *pin) { this->pin_s1_ = pin; }
+  /// set the pin connected to multiplexer control pin 2
+  void set_pin_s2(InternalGPIOPin *pin) { this->pin_s2_ = pin; }
+  /// set the pin connected to multiplexer control pin 3
+  void set_pin_s3(InternalGPIOPin *pin) { this->pin_s3_ = pin; }
+
+  /// set the delay needed after an input switch
+  void set_switch_delay(uint32_t switch_delay) { this->switch_delay_ = switch_delay; }
+
+ private:
+  InternalGPIOPin *pin_s0_;
+  InternalGPIOPin *pin_s1_;
+  InternalGPIOPin *pin_s2_;
+  InternalGPIOPin *pin_s3_;
+  /// the currently active pin
+  uint8_t active_pin_;
+  uint32_t switch_delay_;
+};
+
+class CD74HC4067Sensor : public sensor::Sensor, public PollingComponent, public voltage_sampler::VoltageSampler {
+ public:
+  CD74HC4067Sensor(CD74HC4067Component *parent, const std::string &name, uint8_t pin);
```

Comment:
Please remove the name and pin from the constructor and use the `set_pin` setter function below from `to_code` too.

Could also remove the constructor definition altogether and extend `public Parented<CD74HC4067Component>` too and use `await cg.register_parented(var, config[CONF_CD74HC4067_ID])`

```suggestion
  CD74HC4067Sensor(CD74HC4067Component *parent);
```

Reply:
changed